db/substate_db.go

```diff
@@ -108,12 +110,12 @@ func (db *substateDB) GetSubstate(block uint64, tx int) (*substate.Substate, err
 		return nil, fmt.Errorf("cannot get substate block: %v, tx: %v from db; %w", block, tx, err)
 	}
 
-	rlpSubstate, err := rlp.Decode(val)
-	if err != nil {
-		return nil, fmt.Errorf("cannot decode data into rlp block: %v, tx %v; %w", block, tx, err)
+	pbSubstate := &pb.Substate{}
```

Comment:
Are we swapping to `protobuff` even with current data?

Reply:
This branch was supposed to be a dev branch where protobuf replaces rlp.
The other flag branch introduces the flag to switch between the two.

Now that I think about it, introducing the flag as a standalone first before introducing protobuf would be a cleaner way to package these PRs.

Reply:
Ye, since you are merging to main, we need to prevail `rlp` as default before we know `protobuf` 100% works.

Reply:
Refactored into flag -> protobuf rather than the other way around.

Before:
https://github.com/Fantom-foundation/Substate/pull/85 -> https://github.com/Fantom-foundation/Substate/pull/87

After
https://github.com/Fantom-foundation/Substate/pull/88 -> https://github.com/Fantom-foundation/Substate/pull/85